src/tests/src/solver/optim/api/testApiOrtoolsLinearProblem.cpp

```diff
@@ -0,0 +1,242 @@
+    BOOST_CHECK_THROW(pb->getConstraint(""), std::out_of_range);
+}
+
+BOOST_FIXTURE_TEST_CASE(maximizeMinimize, Fixture)
+{
+    pb->setMinimization();
+    BOOST_CHECK(pb->isMinimization());
+
+    pb->setMaximization();
+    BOOST_CHECK(pb->isMaximization());
+}
+
+BOOST_FIXTURE_TEST_CASE(mipVariableBounds, Fixture)
+{
+    auto* var = pb->addNumVariable(0, 1, "a");
+
+    var->setLb(-4);
+    var->setUb(7);
+
+    BOOST_CHECK_EQUAL(var->getLb(), -4);
+    BOOST_CHECK_EQUAL(var->getUb(), 7);
+
+    var->setBounds(2, 13);
+
+    BOOST_CHECK_EQUAL(var->getLb(), 2);
+    BOOST_CHECK_EQUAL(var->getUb(), 13);
+
+    auto* sameVar = pb->getVariable("a");
+    BOOST_CHECK_EQUAL(sameVar->getLb(), 2);
+    BOOST_CHECK_EQUAL(sameVar->getUb(), 13);
+}
+
+BOOST_FIXTURE_TEST_CASE(mipConstraintBounds, Fixture)
+{
+    auto* constraint = pb->addConstraint(0, 1, "a");
+
+    constraint->setLb(-4);
+    constraint->setUb(7);
+
+    BOOST_CHECK_EQUAL(constraint->getLb(), -4);
+    BOOST_CHECK_EQUAL(constraint->getUb(), 7);
+
+    constraint->setBounds(2, 13);
+
+    BOOST_CHECK_EQUAL(constraint->getLb(), 2);
+    BOOST_CHECK_EQUAL(constraint->getUb(), 13);
+}
+
+BOOST_FIXTURE_TEST_CASE(objectiveCoeff, Fixture)
+{
+    auto* var = pb->addVariable(0, 1, true, "a");
+    pb->setObjectiveCoefficient(var, 1);
+    BOOST_CHECK_EQUAL(pb->getObjectiveCoefficient(var), 1);
+
+    BOOST_CHECK_THROW(pb->getObjectiveCoefficient(nullptr), std::bad_cast);
+    BOOST_CHECK_THROW(pb->setObjectiveCoefficient(nullptr, 0), std::bad_cast);
+}
+
+BOOST_FIXTURE_TEST_CASE(infeasibleProblem, Fixture)
+{
+    auto* a = pb->addIntVariable(0, 10, "a");
+    auto* b = pb->addNumVariable(0, 10, "b");
+    auto* c = pb->addConstraint(1, 1, "c");
+
+    c->setCoefficient(a, 1);
+    c->setCoefficient(b, 1);
+
+    pb->setObjectiveCoefficient(a, 1);
+    pb->setObjectiveCoefficient(b, 1);
+    pb->setMinimization();
+
+    auto* solution = pb->solve(true);
+
+    BOOST_CHECK(solution->getStatus() == Api::MipStatus::MIP_ERROR);
+
+    BOOST_CHECK_EQUAL(solution->getObjectiveValue(), 0);
+    BOOST_CHECK_EQUAL(solution->getOptimalValue(a), 0);
+}
+
+BOOST_FIXTURE_TEST_CASE(problemMaximize, Fixture)
+{
+    createProblemMaximize();
+    auto* solution = pb->solve(false);
+
+    BOOST_CHECK(solution->getStatus() == Api::MipStatus::OPTIMAL);
+
+    BOOST_CHECK_EQUAL(solution->getObjectiveValue(), 1);
+}
+
+BOOST_FIXTURE_TEST_CASE(solutionOpimalValues, Fixture)
+{
+    createProblemMaximize();
+    auto* solution = pb->solve(true);
+
+    auto* a = pb->getVariable("a");
+    BOOST_CHECK_EQUAL(solution->getOptimalValue(a), 1);
+
+    auto* b = pb->getVariable("b");
+
+    std::vector<Api::IMipVariable*> v = {a, b};
+    auto res = solution->getOptimalValues(v);
+
+    auto* varNotInSolution = pb->addNumVariable(0, 1, "f");
+    BOOST_CHECK_EQUAL(solution->getOptimalValue(varNotInSolution), 0);
+
+    BOOST_CHECK_EQUAL(solution->getOptimalValue(nullptr), 0);
+}
+
+BOOST_AUTO_TEST_CASE(simpleProblem)
```

Comment:
unit test : **simpleProblem**

There is no test here.
This makes this "test" useless, as as it won't never raise an testing error.

Reply:
Removed in #2377 and merged here